src/com/vinnik/kate/MultithreadingSorter.java

```diff
@@ -0,0 +1,56 @@
+package com.vinnik.kate;
+
+import java.util.Random;
+import java.util.concurrent.TimeUnit;
+
+/** Class, which creates given number of threads, and measures time they need to sort random array. */
+public class MultithreadingSorter {
+
+    /** Creates given number of threads and executes them. */
+    public static void measureTimeOfSorting(int amountOfThreads) {
+        for (int i = 0; i < amountOfThreads; i++) {
+            ThreadSort thread = new ThreadSort(i);
+            thread.start();
+        }
+    }
+
+
+    private static class ThreadSort extends Thread {
+        private int[] numbersToSort;
+        private int key;
+
+        protected ThreadSort(int number) {
+            numbersToSort = new int[500];
+            key = number;
+
+            for (int i = 0; i < numbersToSort.length; i++) {
+                Random random = new Random();
```

Comment:
I don't think it's sufficient to create a new Random instance in each iteration.
Instead, you could create `final Random random = ...` instance **before** the loop and use it to generate random numbers.
Also, a good practice is to keep constructor's code as clear as possible. Consider move the array generation into a separate function. I'd do something like `this. numbersToSort = this.getRandomArray(number)`, where `this.getRandomArray(...)` is the function to generate random numbers.